javalib/src/main/scala/java/util/HashSet.scala

```diff
@@ -0,0 +1,89 @@
+package java.util
+
+import scala.collection.mutable
+import scala.collection.JavaConversions._
+
+class HashSet[E](initialCapacity: Int, loadFactor: Float) 
+    extends java.util.AbstractSet[E] with java.util.Set[E] with Cloneable with Serializable {
+
+  def this() =
+  	this(16, 0.75F)
+
+  def this(initialCapacity: Int) =
+  	this(initialCapacity, 0.75F)
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  protected val inner: mutable.Map[Int, E] = new scala.collection.mutable.HashMap[Int, E]()
+
+  override def contains(o: Any): Boolean =
+    inner.contains(o.hashCode)
```

Comment:
You should use the same `Box` trick. This suggest that `Box` should be a `private[util]` class, instead of buried in `ConcurrentHashMap`.